src/Features/VisualBasic/Portable/CodeRefactorings/NodeSelectionHelpers.vb

```diff
@@ -2,24 +2,49 @@
     Friend Module NodeSelectionHelpers
-        Friend Async Function GetSelectedMemberDeclarationAsync(context As CodeRefactoringContext) As Task(Of SyntaxNode)
-            Dim methodMember = Await context.TryGetRelevantNodeAsync(Of MethodBaseSyntax)().ConfigureAwait(False)
-            If methodMember IsNot Nothing Then
-                Return methodMember
+        Friend Async Function GetSelectedMemberDeclarationAsync(context As CodeRefactoringContext) As Task(Of ImmutableArray(Of SyntaxNode))
+            Dim doc As Document
+            Dim span As TextSpan
+            Dim cancellationToken As CancellationToken
+            context.Deconstruct(doc, span, cancellationToken)
+            If span.IsEmpty Then
+                ' MethodBaseSyntax also includes properties
+                Dim methodMember = Await context.TryGetRelevantNodeAsync(Of MethodBaseSyntax)().ConfigureAwait(False)
+                If methodMember IsNot Nothing Then
+                    Return ImmutableArray.Create(DirectCast(methodMember, SyntaxNode))
+                End If
+                ' Gets field variable declarations (not including the keywords like Public/Shared, etc), which are not methods
+                Dim fieldDeclaration = Await context.TryGetRelevantNodeAsync(Of FieldDeclarationSyntax).ConfigureAwait(False)
+                If fieldDeclaration Is Nothing Then
+                    ' Gets the identifier + type of the field itself (ex. TestField As Integer), since it is nested in the variable declaration
+                    ' And so the token's parent is not a variable declaration
+                    Dim modifiedIdentifier = Await context.TryGetRelevantNodeAsync(Of ModifiedIdentifierSyntax).ConfigureAwait(False)
+                    If modifiedIdentifier Is Nothing Then
+                        Return ImmutableArray(Of SyntaxNode).Empty
+                    Else
+                        Return ImmutableArray.Create(DirectCast(modifiedIdentifier, SyntaxNode))
+                    End If
+                Else
+                    ' Field declarations can contain multiple variables (each of which are a "member")
+                    Return fieldDeclaration.Declarators.SelectMany(Function(vds) vds.Names).Cast(Of SyntaxNode).AsImmutable()
+                End If
+            Else
+                ' if the span is non-empty, then we get potentially multiple members
+                ' Note: even though this method handles the empty span case, we don't use it because it doesn't correctly
+                ' pick up on keywords before the declaration, such as "public static int".
+                ' We could potentially use it for every case if that behavior changes
+                Dim tree = Await doc.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(False)
+                Return Await VisualBasicSelectedMembers.Instance.GetSelectedMembersAsync(tree, span, True, cancellationToken).ConfigureAwait(False)
```

Comment:
The reason we do separate things for different spans is because for nonempty spans, `GetRelevantNodesAsync` will get the largest node that completely encompasses the span and will not get smaller ones. For example, if we selected multiple members in a class, `GetRelevantNodesAsync` gets the class declaration instead of the selected members. However, the `AbstractSelectedMembers` logic will only activate on an empty span if the span is on the identifier name (and not any of the keywords, which we want to include as well). It may be possible to change `AbstractSelectedMembers` to get nodes when the keyword is selected too, but this would end up probably being logic very similar to `GetRelevantNodesAsync`, and would also introduce a feature change in the encapsulate fields/properties and generate method refactorings as well.